web-registry/src/components/organisms/MediaForm/MediaForm.tsx

```diff
@@ -1,92 +1,105 @@
+import Form from 'components/molecules/Form/Form';
+import { useZodForm } from 'components/molecules/Form/hook/useZodForm';
 
-import { PHOTO_COUNT } from './MediaForm.constants';
-import type { MediaErrorsSimple, MediaValuesSimple } from './MediaFormSimple';
+import { DEFAULT_URL } from './MediaForm.constants';
+import { mediaFormSchema, MediaFormSchemaType } from './MediaForm.schema';
+import { validateMediaFormForm } from './MediaForm.utils';
 import { MediaFormSimple } from './MediaFormSimple';
 
-export interface MediaBaseValues {
-  'regen:previewPhoto'?: string;
-  'regen:galleryPhotos'?: Array<string>;
-  'regen:storyMedia'?: ProjectStoryMedia;
-}
-
-export interface MediaBaseErrors {
-  'regen:previewPhoto'?: string;
-  'regen:storyMedia'?: string;
-}
-
-export type MediaFormValues = MediaValuesSimple;
-export type MediaErrors = MediaErrorsSimple;
-
-export const cropAspect = { aspect: 322 / 211 }; // px values pulled from mockups (width / height)
-
 interface MediaFormProps {
-  submit: ({ values }: { values: MediaFormValues }) => Promise<void>;
+  submit: ({ values }: { values: MediaFormSchemaType }) => Promise<void>;
   onPrev?: () => void;
   onNext?: () => void;
-  initialValues: MediaFormValues;
+  initialValues: MediaFormSchemaType;
   projectId?: string;
 }
 
-const MediaFormSchema = Yup.object().shape({
-  'regen:previewPhoto': Yup.string(),
-  'regen:galleryPhotos': Yup.array(Yup.string()).max(PHOTO_COUNT),
-});
-
-/** Formik Context + handlers for legacy and new media */
 export const MediaForm = ({
   initialValues,
   projectId,
-  ...props
+  submit,
+  onNext,
+  onPrev,
 }: MediaFormProps): JSX.Element => {
-  const { confirmSave, isEdit, formRef } = useProjectEditContext();
+  const form = useZodForm({
+    schema: mediaFormSchema,
+    defaultValues: {
+      ...initialValues,
+    },
+    mode: 'onBlur',
+  });
+  const { isSubmitting, isDirty, isValid } = useFormState({
+    control: form.control,
+  });
+  const fileNamesToDeleteRef = useRef<string[]>([]);
+  const { isDirtyRef } = useProjectEditContext();
 
-  async function handleSubmit(
-    values: MediaFormValues,
-    { setSubmitting, setTouched }: FormikHelpers<MediaFormValues>,
-  ): Promise<void> {
-    try {
-      await props.submit({ values });
-      setTouched({}); // reset to untouched
-      if (isEdit && confirmSave) confirmSave();
-    } catch (e) {
-      setSubmitting(false);
-    }
-  }
+  const { confirmSave, isEdit } = useProjectEditContext();
+  const setErrorBannerTextAtom = useSetAtom(errorBannerTextAtom);
 
   return (
     <OnBoardingCard>
-      <Formik
-        innerRef={formRef as FormRef<MediaFormValues>}
-        enableReinitialize
-        validateOnMount
-        initialValues={initialValues}
-        validationSchema={MediaFormSchema}
-        onSubmit={handleSubmit}
+      <Form
+        form={form}
+        onSubmit={async data => {
+          const hasError = validateMediaFormForm({
```

Comment:
```suggestion
          const hasError = validateMediaForm({
```

why do we need the extra validation step btw? this means we need to have the validation logic in two formats: declarative form with zod + validate function
isn't there a way to perform the validation based on zod schema before submitting? running validation on submit is the default behavior for formik based forms for instance: https://formik.org/docs/guides/validation#when-does-validation-run

Reply:
We needed the extra validation at the submit step here because of the empty input field that messes with the validation count. I'll see if I can merge them into refine rules.